bite-sized-bears/Website/syndication_app/views.py

```diff
@@ -0,0 +1,335 @@
+import imghdr
+from functools import update_wrapper
+
+from django.views import View as ParentView
+from django.db.models import Count
+from django.views.generic import ListView
+from django.contrib.auth.models import auth
+from django.shortcuts import render, redirect
+from django.views.generic.edit import UpdateView, CreateView
+from django.contrib.auth.decorators import login_required
+from django.utils.decorators import classonlymethod
+from django.contrib.auth.models import User as AuthUser
+
+from .models import Post, Comments, User, Community
+
+
+class View(ParentView):
+    """
+    Custom View Class for extending the django User Model
+    """
+    @classonlymethod
+    def as_view(cls, **initkwargs):
+        for key in initkwargs:
+            if key in cls.http_method_names:
+                raise TypeError("You tried to pass in the %s method name as a "
+                                "keyword argument to %s(). Don't do that."
+                                % (key, cls.__name__))
+            if not hasattr(cls, key):
+                raise TypeError("%s() received an invalid keyword %r. as_view "
+                                "only accepts arguments that are already "
+                                "attributes of the class." % (cls.__name__, key))
+
+        def view(request, *args, **kwargs):
+            self = cls(**initkwargs)
+            try:
+                request.user = User.objects.get(name=request.user.get_username())
+                auth_user = AuthUser.objects.get(username=request.user.name)
+                for meth in dir(auth_user):
+                    if not hasattr(request.user, meth):
+                        try:
+                            setattr(request.user, meth, getattr(auth_user, meth))
+                        except Exception:
+                            pass
+            except Exception:
+                pass
+            if hasattr(self, 'get') and not hasattr(self, 'head'):
+                self.head = self.get
+            self.setup(request, *args, **kwargs)
+            if not hasattr(self, 'request'):
+                raise AttributeError(
+                    "%s instance has no 'request' attribute. Did you override "
+                    "setup() and forget to call super()?" % cls.__name__
+                )
+            return self.dispatch(request, *args, **kwargs)
+
+        view.view_class = cls
+        view.view_initkwargs = initkwargs
+
+        update_wrapper(view, cls, updated=())
+
+        update_wrapper(view, cls.dispatch, assigned=())
+        return view
+
+
+class IndexListView(ListView, View):
+    paginate_by = 25
```

Comment:
`IndexListView` has 25 items per page, while `CommunityListView` has 20 items, and `MostViewedPost` and `MyCommunityListView` have 15 items. A comment with the rationale for having varied pagination sizes would be good.
Alternatively, a constant can be used to unify pagination sizes, eg. `PAGINATION_SIZE = 20`.